src/authorizedApiContext.ts

```diff
@@ -592,6 +595,40 @@ export class AuthorizedApiContext {
     };
   }
 
+  /**
+   * Returns the size units that match the given constraints.
+   * @param preparedTrade the result from the prepare trade api call
+   * @param orderModel the currently selected order model in the order form
+   * @param direction the currently selected direction in the order form
+   * @param cashAccountId the id of the currently selected cash account in the order form
+   */
+  _getSizeUnitFromConstraints(
+    preparedTrade: PreparedTrade,
+    orderModel: OrderModel,
+    direction: Direction,
+    cashAccountId: string
+  ) {
+    for (const constraint of preparedTrade.sizeUnitConstraints || []) {
+      if (
+        !constraint.orderModels ||
+        constraint.orderModels.includes(orderModel)
+      ) {
+        if (
+          !constraint.directions ||
+          constraint.directions.includes(direction)
+        ) {
+          if (
+            !constraint.cashAccountIds ||
+            constraint.cashAccountIds.includes(cashAccountId)
+          ) {
+            return constraint.sizeUnits;
+          }
+        }
+      }
+    }
+    return [];
```

Comment:
Als Fallback gerne einfach (falls != null) `[preparedTrade.sizeUnit]` ausgeben - das wird glaube ich auch nach wie vor bei den meisten Brokern anstelle der constraints gesetzt (alternativ im backend bei *jedem* Broker ein `sizeUnitConstraints` implementieren).

Reply:
öhh sicher? 
das ist ja explizit ne Funktion die aus den `sizeUnitConstraints` die units rauszieht und nicht generell alle sizeUnits durchgeht. 
im Frontend bei uns ist das auch nochmal gewrappt in den `buildSizeUnitOptions(...)`

Reply:
Ich würde es so sehen: die Funktion liefert die auswählbaren `sizeUnits` zu der gegebenen Kombination aus `preparedTrade`, `orderModel`, `direction` und `cashAccountId`. Falls ein `PreparedTrade` *keine* `sizeUnitConstraints` implementiert, dann ist das Ergebnis `[preparedTrade.sizeUnit]`. Somit braucht man im Frontend dann auch nur einen einzigen Weg zu implementieren.

Reply:
ok ja passt. Ich hab auch vergessen dass `perparedTrade.sizeUnitsByCashAccountId` deprecated wird.